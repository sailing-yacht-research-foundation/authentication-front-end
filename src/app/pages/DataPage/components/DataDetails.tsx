import React from 'react';
import styled from 'styled-components';
import Iframe from 'react-iframe'
import { StyleConstants } from 'styles/StyleConstants';
import { IoIosArrowBack } from 'react-icons/io';
import { GobackButton, PageDescription, PageHeading } from 'app/components/SyrfGeneral';

export const DataDetails = (props) => {
    const { data, goBack } = props;

    React.useEffect(() => {
        window.scroll(0, 0);
    }, []);

    return (
        <Wrapper>
            <PageHeadContainer>
                <GobackButton onClick={() => goBack()}>
                    <IoIosArrowBack style={{fontSize:'40px', color: '#1890ff'}}/>
                </GobackButton>
                <PageInfoContainer>
                    <PageHeading>{data.title}</PageHeading>
                    <PageDescription>{data.description}</PageDescription>
                </PageInfoContainer>
            </PageHeadContainer>
            <IframeContainer>
                <Iframe width={'100%'} height={`100%`} url={data.htmlUrl}></Iframe>
            </IframeContainer>
        </Wrapper>
    )
}


const PageHeadContainer = styled.div`
    display: flex;
    align-items: center;
`;

const PageInfoContainer = styled.div`
    display: flex;
    flex-direction: column;
`;

const Wrapper = styled.div`
 margin-top: ${StyleConstants.NAV_BAR_HEIGHT};
`;

const IframeContainer = styled.div`
    height: calc(${window.innerHeight}px - 17vh);
`;