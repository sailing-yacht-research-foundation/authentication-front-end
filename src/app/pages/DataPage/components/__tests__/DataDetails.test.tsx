import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { DataDetails } from '../DataDetails';

const shallowRenderer = createRenderer();

describe('<DataDetails />', () => {
    it('should render the deals page and matchs the snapshot', () => {
        shallowRenderer.render(<DataDetails data={
            {
                title: 'Yacht Club Density',
                description: '3D Hexbin map to visualize the density of yacht clubs.',
                screenshotUrl: '/static-htmls/E_yacht_clubs/screenshot.png',
                htmlUrl: '/static-htmls/E_yacht_clubs/yacht_clubs.html',
            }
        } />);
        const renderedOutput = shallowRenderer.getRenderOutput();
        expect(renderedOutput).toMatchSnapshot();
    });
});