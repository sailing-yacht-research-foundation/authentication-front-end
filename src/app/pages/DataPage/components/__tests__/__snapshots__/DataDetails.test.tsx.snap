// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataDetails /> should render the deals page and matchs the snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div
      onClick={[Function]}
    >
      <IoIosArrowBack
        style={
          Object {
            "color": "#1890ff",
            "fontSize": "40px",
          }
        }
      />
    </styled.div>
    <styled.div>
      <styled.h2>
        Yacht Club Density
      </styled.h2>
      <styled.p>
        3D Hexbin map to visualize the density of yacht clubs.
      </styled.p>
    </styled.div>
  </styled.div>
  <styled.div>
    <Iframe
      height="100%"
      url="/static-htmls/E_yacht_clubs/yacht_clubs.html"
      width="100%"
    />
  </styled.div>
</styled.div>
`;
